internal/provider/resource_gitlab_project_test.go

```diff
@@ -1139,6 +1139,52 @@ func TestAccGitlabProject_templateMutualExclusiveNameAndID(t *testing.T) {
 	})
 }
 
+// Gitlab update project API call requires one from a subset of project fields to be set (See #1157)
+// If only a non-blessed field is changed, this test checks that the provider ensures the code won't return an error.
+func TestAccGitlabProject_UpdateAnalyticsAccessLevel(t *testing.T) {
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabProjectDestroy,
+		Steps: []resource.TestStep{
+			// Create minimal test project
+			{
+				Config: fmt.Sprintf(`
+					resource "gitlab_project" "this" {
+						name = "foo-%d"
+						visibility_level                = "public"
+						analytics_access_level = "private"
+					}`, rInt),
+			},
+			// Verify Import
+			{
+				ResourceName:      "gitlab_project.this",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// Update `analytics_access_level`
+			{
+				Config: fmt.Sprintf(`
+					resource "gitlab_project" "this" {
+						name = "foo-%d"
+						visibility_level = "public"
+						analytics_access_level = "disabled"
+					}`, rInt),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("gitlab_project.this", "analytics_access_level", "disabled"),
+				),
```

Comment:
I've missed that last time: because it's the only check left we don't need the compose.

```suggestion
				resource.TestCheckResourceAttr("gitlab_project.this", "analytics_access_level", "disabled"),
```

Reply:
Do we still need the `Check:` though?

Reply:
hmmm, you actually don't :) 👍 The subsequent `Verify Import` will make sure that what's coming from the read API matches the config and also this very step checks if the plan is empty after the apply..

```suggestion
```